### PR TITLE
feature: create Helm chart for a pure in-mem runtime

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## WHAT

this PR adds a new Helm Chart `tractusx-connector-memory`, that uses only in-memory components (save for DAPS). Furthermore, `edc-controlplane-memory` was renamed to `edc-runtime-memory`, and I removed the Azure KeyVault from it. 
I also created a new folder for the test infrastructure, which is a pruned-down version of the "old" `supporting-infrastructure`

## WHY

Could be useful for testing or development purposes. lay the ground work for deploy-testing the production charts (future PRs)

## FURTHER NOTES
- added an in-memory vault, that can be seeded with secrets upon startup. 
- moved the `run-deployment-test` workflow to a composite action
- used `helm test` to ping the runtime's health endpoint
- obviously, more elaborate tests can be added, such as sending `POST` requests against the management api, and verifying that an object was actually created, etc.

## OPEN QUESTIONS
- seed secrets are stored in `.github/workflows/deploy-test-secrets`. is that the proper place?

### CODE REVIEW

File: .github/actions/publish-docker-image/action.yml
Comment (florianrusch-zf) [inline .github/actions/publish-docker-image/action.yml:1]: Why are these changes necessary?

File: .github/actions/run-deployment-test/action.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    - name: Setup Kubectl
+      uses: azure/setup-kubectl@v3.2
+
+    - name: Create k8s Kind Cluster
+      uses: helm/kind-action@v1.5.0
+
+    - name: Load images into KinD
+      shell: bash
+      run: |
+        kind get clusters | xargs -n1 kind load docker-image ${{ inputs.imagename}}:${{ inputs.image_tag}} --name
```
Comment (florianrusch-zf) [inline .github/actions/run-deployment-test/action.yml:80]:
```suggestion
        kind get clusters | xargs -n1 kind load docker-image ${{ inputs.imagename }}:${{ inputs.image_tag }} --name
```

File: .github/actions/run-deployment-test/action.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+    #################
+    ### Tear Down ###
+    #################
+    - name: Destroy the kind cluster
+      if: always()
+      shell: bash
+      run: >-
+        kind get clusters | xargs -n1 kind delete cluster --name
+
```
Comment (florianrusch-zf) [inline .github/actions/run-deployment-test/action.yml:104]:
```suggestion

```

File: .github/actions/run-deployment-test/action.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      shell: bash
+      run: |-
+        helm install infra edc-tests/deployment/src/main/resources/helm/test-infrastructure \
+        --wait-for-jobs --timeout=30s --dependency-update
+
+    - name: Install In-Mem Runtime
+      shell: bash
+      run: ${{ inputs.helm_command }}
+
+
```
Comment (florianrusch-zf) [inline .github/actions/run-deployment-test/action.yml:95]:
```suggestion
```

File: .github/dependabot.yml [hunk 4]
```diff
@@ -2,8 +2,7 @@
 version: 2
 updates:
   # Maven
-  -
-    package-ecosystem: "maven"
+  - package-ecosystem: "maven"
```
Comment (florianrusch-zf) [inline .github/dependabot.yml:5]: This needs to be replaced with gradle, right? Do you know if we have an issue for that already?

File: settings.gradle.kts [hunk 5]
```diff
@@ -3,6 +3,7 @@ rootProject.name = "tractusx-edc"
 include(":edc-extensions:business-partner-validation")
 include(":edc-extensions:control-plane-adapter")
 include(":edc-extensions:cx-oauth2")
+include(":edc-extensions:vault-memory")
```
Comment (florianrusch-zf) [inline settings.gradle.kts]: Yes, I would highly agree to move it into the `edc-runtime-memory` module. If we keep it as an extension we would also publish this as a jar which mean people will maybe us it, which again means we need to maintain/support/update in a good shape. Of course we also need to maintain/support/update the `edc-runtime-memory` but I think it's less work for us

File: settings.gradle.kts [hunk 6]
```diff
@@ -138,37 +139,37 @@ dependencyResolutionManagement {
             library("micrometer-jetty", "org.eclipse.edc", "jetty-micrometer").versionRef("edc")
             library("monitor-jdklogger", "org.eclipse.edc", "monitor-jdk-logger").versionRef("edc")
             library(
-                "transfer.dynamicreceiver",
-                "org.eclipse.edc",
-                "transfer-pull-http-dynamic-receiver"
+                    "transfer.dynamicreceiver",
```
Comment (florianrusch-zf) [inline settings.gradle.kts:141]: The indention looks weird for me. 8 spaces? Is this correct?

File: edc-tests/deployment/src/main/resources/helm/test-infrastructure/values.yaml [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+        LXRWZjHwexDqtgMGTCN/7aZlMA8GA1UdEwEB/wQFMAMBAf8wDQYJKoZIhvcNAQEL
+        BQADggEBAD2a5kuIdICNXfYLpSe7AIONwZVucaArYtpXBxHEy5lMJsTEJgjZzypd
+        iIMU7onEQGVbii6yVNpWfIpJYM4e8ytVdJuk5evclVKZs/lZ2IshLyWFVj+ITh2E
+        28X4C/Hnmt4MPBCNowQf71nMp4LEziBgXp54qFV9C+qSTEVdrherRE0PU/zKyX10
+        S/P5o42weTHnAO/pBN/8AmL3AymynKVgcPaW46IjjRAuc6kfZWCrYQ0M4+/7Ws5r
+        uM55Zae/L+C82OTNNaaK324ogsCkORPeQ23OCrRD8rZJmQ9bpoOGglPminfwEOhB
+        UHtyKgmvqCyOV3G/4G93W/xsLV0kxLA=
+        -----END CERTIFICATE-----
+        EOF
+        }
```
Comment (florianrusch-zf) [inline edc-tests/deployment/src/main/resources/helm/test-infrastructure/values.yaml:185]:
```suggestion
        }

```

File: edc-tests/deployment/src/main/resources/helm/test-infrastructure/README.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+> risk! ([source](https://github.com/Fraunhofer-AISEC/omejdn-server))
+
+### HashiCorp Vault
+
+The Control- and Data Plane persist confidential in the vault and persist and communicate using only the secret
+names. Hence, it is not possible to run a connector without an instance of a vault.
+
+### Backend Application
+
+After a Data Transfer is successfully prepared the control plane will contact the a configurable endpoint with the
```
Comment (florianrusch-zf) [inline edc-tests/deployment/src/main/resources/helm/test-infrastructure/README.md]:
```suggestion
After a Data Transfer is successfully prepared the control plane will contact the configurable endpoint with the
```

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]
just a few comments already - hard to review if you are still pushing things 😅 Will try to review again at the weekend.

File: edc-tests/deployment/src/main/resources/helm/test-infrastructure/README.md (hunk 8 above)
Comment (paullatzelsperger) [reply edc-tests/deployment/src/main/resources/helm/test-infrastructure/README.md]: in the meantime i removed this section

File: settings.gradle.kts (hunk 5 above)
Comment (paullatzelsperger) [reply settings.gradle.kts]: yep, already done

File: .github/actions/publish-docker-image/action.yml
Comment (paullatzelsperger) [reply .github/actions/publish-docker-image/action.yml:1]: i just came across them and fixed them. a composite action should not care about the trigger event or execution context

File: .github/dependabot.yml (hunk 4 above)
Comment (paullatzelsperger) [reply .github/dependabot.yml:5]: > just a few comments already - hard to review if you are still pushing things 😅 Will try to review again at the weekend.

yeah, i'm sorry, just went through the code, and noticed a couple of things, which i pushed right away ... :/

Comment (paullatzelsperger) [reply .github/dependabot.yml:5]: i don't think we haven issue, will create one

File: settings.gradle.kts (hunk 6 above)
Comment (paullatzelsperger) [reply settings.gradle.kts:141]: nope, thx

File: README.md [hunk 9]
```diff
@@ -16,24 +16,28 @@ Please also refer to:
...
+The eclipse data space connector is split up into Control-Plane and Data-Plane, whereas the Control-Plane functions as
+administration layer
 and has responsibility of resource management, contract negotiation and administer data transfer.
 The Data-Plane does the heavy lifting of transferring and receiving data streams.
 
 Depending on your environment there are different derivatives of the control-plane prepared:
 
-- [edc-controlplane-memory](edc-controlplane/edc-controlplane-memory) with dependency onto
+- [edc-runtime-memory](edc-controlplane/edc-runtime-memory) with dependency onto
   - [Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/#product-overview)
```
Comment (wolf4ood) [inline README.md:33]: This deps is not valid anymore right?

File: docs/README.md [hunk 10]
```diff
@@ -13,7 +15,7 @@ When running a EDC connector from the Tractus-X EDC repository there are three s
 The three supported setups are.
 
 - Setup 1: In Memory & Azure Vault
-  - [Control Plane](../edc-controlplane/edc-controlplane-memory/README.md)
+  - [Control Plane](../edc-controlplane/edc-runtime-memory/README.md)
     - [IDS DAPS Extensions](https://github.com/eclipse-edc/Connector/tree/main/extensions/common/iam/oauth2/daps)
     - In Memory Persistence done by using no extension
     - [Azure Key Vault Extension](https://github.com/eclipse-edc/Connector/tree/main/extensions/common/vault/azure-vault)
```
Comment (wolf4ood) [inline docs/README.md:21]: Also this one probably the dep to the vault  it's not valid anymore

File: edc-controlplane/edc-runtime-memory/src/main/java/org/eclipse/tractusx/edc/vault/memory/VaultMemoryExtension.java [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+@Extension(value = "In-memory vault extension", categories = {"vault", "security"})
+public class VaultMemoryExtension implements ServiceExtension {
+
+    @Setting(value = "Secrets with which the vault gets initially populated. Specify as comma-separated list of key:secret pairs.")
+    public static final String VAULT_MEMORY_SECRETS_PROPERTY = "edc.vault.secrets";
+
+    @Override
+    public String name() {
+        return "In-Memory Vault Extension";
+    }
```
Comment (wolf4ood) [inline edc-controlplane/edc-runtime-memory/src/main/java/org/eclipse/tractusx/edc/vault/memory/VaultMemoryExtension.java:36]:
```suggestion
@Extension(value = VaultMemoryExtension.NAME, categories = {"vault", "security"})
public class VaultMemoryExtension implements ServiceExtension {

    public static final String NAME = "In-Memory Vault Extension";

    @Setting(value = "Secrets with which the vault gets initially populated. Specify as comma-separated list of key:secret pairs.")
    public static final String VAULT_MEMORY_SECRETS_PROPERTY = "edc.vault.secrets";

    @Override
    public String name() {
        return NAME;
    }
```

**Review by wolf4ood — COMMENTED** [review]
Just a nit I would store the name on a static var

**Comment by paullatzelsperger** [comment]
had to close, will re-open

File: .github/dependabot.yml (hunk 4 above)
Comment (paullatzelsperger) [reply .github/dependabot.yml:5]: [edit] #216
